cpp/src/arrow/array-decimal-test.cc

```diff
@@ -75,8 +75,7 @@ class DecimalTest : public DecimalTestBase<T> {
  public:
   std::vector<uint8_t> data(
       const std::vector<T>& input, size_t byte_width) const override {
-    std::vector<uint8_t> result;
-    result.reserve(input.size() * byte_width);
+    std::vector<uint8_t> result(input.size() * byte_width);
```

Comment:
It is called - since result's buffer is allocated by reserve() it works, but msvc debug run-time complaints on the access to zero-size vector.